src/sources/journald.rs

```diff
@@ -174,6 +178,16 @@ impl JournaldConfig {
             .boxed()
             .compat()
             .map_err(|error| error!(message="Journald server unexpectedly stopped.",%error))
+            .select(shutdown.map(|_| ()))
+            .map(move |_| {
+                if let Some(pid) = pid {
+                    // Signal the child process to terminate so that the
+                    // blocking future can be unblocked earlier rather
+                    // than later.
+                    let _ = kill(pid, Signal::SIGTERM);
```

Comment:
Good thinking, definitely have to deal with this. However, could this be part of the `JournalSource` trait, to let the source itself handle the kill instead of handling it externally? That would eliminate the need to handle the PID outside of `struct Journalctl`.

Reply:
PID can be moved, but something still needs to be passed back to trigger the kill. So what about this with passing `FnOnce()`? It is nicer.

Reply:
I see the `journald_server` ends up being moved into the spawned block, and has a `&mut self` method, which makes sharing it overly difficult. That is probably the best compromise.